packages/query/src/query-engine/index.ts

```diff
@@ -62,4 +65,95 @@ export class QueryEngine {
 
 		return parseTemplate( data, wpHomeUrl, uri );
 	};
+
+	/**
+	 * Fetches paths to be rendered statically.
+	 *
+	 * @param {GetPagesToRenderStaticallyQueryVariables} variables - The variables for the query.
+	 * @return The path data to be rendered statically.
+	 */
+	static getStaticPaths = async (
+		variables: GetPagesToRenderStaticallyQueryVariables = {}
+	): Promise< string[] > => {
+		const data = await fetchQuery( {
+			name: 'GetPagesToRenderStatically',
+			query: GetPagesToRenderStaticallyDocument,
+			options: {
+				variables: {
+					...variables,
+				},
+			},
+		} );
+
+		if ( ! data ) {
+			throw new Error(
+				'Error fetching pages to render statically. No data returned.'
+			);
+		}
+
+		const resolvedData: string[] = [];
+		let shouldFetchMore = false;
+		const newVariables = {
+			...variables,
+		};
+
+		Object.entries( data ).forEach( ( [ key, section ] ) => {
+			if ( ! section ) {
+				return;
+			}
+
+			// Here we know for sure that the key which is string is a key of GetPagesToRenderStaticallyQuery so it's safe to cast it.
+			const resolvedKey = key as keyof GetPagesToRenderStaticallyQuery;
+
+			const { nodes, pageInfo } = section as {
+				nodes: StaticRouteNodeDataFragment[];
+				pageInfo: StaticRoutePageInfoFragment;
+			};
+
+			resolvedData.push(
+				...nodes
+					.filter( ( { uri } ) => !! uri )
+					.map( ( { uri } ) => uri! )
+			);
+
+			const cursorKey =
+				`${ resolvedKey }Cursor` satisfies keyof GetPagesToRenderStaticallyQueryVariables;
+			const hasMoreKey = `hasMore${ capitalize(
```

Comment:
Didn't I get rid of this needless `capitalize()` by inverting the query variable name?